scrunch/datasets.py

```diff
@@ -952,9 +987,9 @@ def create_single_response(self, categories,
                                  expr=expr,
                                  description=description))
 
-        return Variable(
-            self.resource.variables.create(payload).refresh(),
-            self.resource)
+        new_var = self.resource.variables.create(payload).refresh()
+        # return the variable instance
+        return self[new_var.body.alias]
```

Comment:
This is not going to work. You have `self._vars` that reflects the old list of variables that does not contain this new variable there, so this will raise a variable not found error.
We need a hook that is part of the Mixin that refreshes the variable list. On the init we'd also call that same method and this will be what we use to invalidate cache.